cedar-policy-core/src/authorizer.rs

```diff
@@ -892,4 +892,8 @@ pub enum Decision {
     /// that no decision could be safely reached; for example, errors parsing
     /// the policies.
     Deny,
+    /// The `Authorizer` determined that some information required to reach
+    /// a decision is missing.
+    #[cfg(feature = "partial-eval")]
+    NoDecision,
```

Comment:
One initial comment from a cursory look at the code: I'm a bit concerned with adding `NoDecision` to this struct. We eventually want to stabilize partial evaluation which would mean removing the feature flag (or enabling it by default), at which point non-partial-eval users would need to know to ignore this variant.

Reply:
One way to avoid this would be to have a completely different interface response format for residual decisions that wouldn't include a decision member at all. Interestingly, [com.cedarpolicy.model.AuthorizationResponse.Decision](https://github.com/cedar-policy/cedar-java/blob/main/CedarJava/src/main/java/com/cedarpolicy/model/AuthorizationResponse.java#L48) in cedar-java/CedarJava already has a NoDecision enum value.

Reply:
That's true, but there is already an open issue to remove the `NoDecision` enum value: https://github.com/cedar-policy/cedar-java/issues/37.  (Obviously this could be reconsidered)

Reply:
I removed the NoDecision and reworked the logic. I think this solution is cleaner. The JSON object for a concrete response is exactly the same as what would be produced by json_is_authorized(). The JSON object produced when residuals policies are returned only contains residuals and diagnostics keys.